src/web/app/pages-instructor/instructor-student-list-page/instructor-student-list-page.component.ts

```diff
@@ -12,13 +69,490 @@ import { ActivatedRoute } from '@angular/router';
 export class InstructorStudentListPageComponent implements OnInit {
 
   user: string = '';
+  isDisplayArchive: boolean = false;
+  courses: Course[] = [];
+
+  searchQuery: string = '';
+
+  courseDetailsList: CourseDetails[] = [];
+  sectionDetailsList: SectionDetails[] = [];
+  teamDetailsList: TeamDetails[] = [];
+  studentDetailsList: StudentDetails[] = [];
+
+  get allPresentCourses(): Course[] {
+    return this.courses.filter((course: Course) => !(course.isArchived && !this.isDisplayArchive));
+  }
+
+  get allCheckedCourses(): Course[] {
+    return this.allPresentCourses.filter((course: Course) => course.isChecked);
+  }
+
+  get allPresentCourseDetails(): CourseDetails[] {
+    const courseDetailsList: CourseDetails[] = [];
+
+    this.allPresentCourses.forEach((course: Course) =>  {
+      const courseDetails: CourseDetails | undefined = this.getCourseDetails(course.id);
+      if (courseDetails) {
+        courseDetailsList.push(courseDetails);
+      }
+    });
+
+    return courseDetailsList;
+  }
+
+  get allCheckedCourseDetails(): CourseDetails[] {
+    return this.allPresentCourseDetails.filter((courseDetails: CourseDetails) => courseDetails.isChecked);
+  }
 
-  constructor(private route: ActivatedRoute) { }
+  get allPresentSectionDetails(): SectionDetails[] {
+    let allPresentSectionDetails: SectionDetails[] = [];
+
+    this.allCheckedCourseDetails.forEach((courseDetails: CourseDetails) => {
+      allPresentSectionDetails = allPresentSectionDetails.concat(courseDetails.sections);
+    });
+
+    return allPresentSectionDetails;
+  }
+
+  get allCheckedSectionDetails(): SectionDetails[] {
+    return this.allPresentSectionDetails.filter((sectionDetails: SectionDetails) => sectionDetails.isChecked);
+  }
+
+  get allPresentTeamDetails(): TeamDetails[] {
+    let allPresentTeamDetails: TeamDetails[] = [];
+
+    this.allCheckedSectionDetails.forEach((sectionDetails: SectionDetails) => {
+      allPresentTeamDetails = allPresentTeamDetails.concat(sectionDetails.teams);
+    });
+
+    return allPresentTeamDetails;
+  }
+
+  get allCheckedTeamDetails(): TeamDetails[] {
+    return this.allPresentTeamDetails.filter((teamDetails: TeamDetails) => teamDetails.isChecked);
+  }
+
+  get allPresentStudentDetails(): StudentDetails[] {
+    let allPresentStudentDetails: StudentDetails[] = [];
+
+    this.allCheckedTeamDetails.forEach((teamDetails: TeamDetails) => {
+      allPresentStudentDetails = allPresentStudentDetails.concat(teamDetails.students);
+    });
+
+    return allPresentStudentDetails;
+  }
+
+  get allCheckedStudentDetails(): StudentDetails[] {
+    return this.allPresentStudentDetails.filter((studentDetails: StudentDetails) => studentDetails.isChecked);
+  }
+
+  get isAllPresentCoursesChecked(): boolean {
+    if (this.allPresentCourses.length === 0) {
+      return false;
+    }
+    return this.allPresentCourses.length === this.allCheckedCourses.length;
```

Comment:
You can shorten this to
```ts
return this.allPresentCourses.length && this.allPresentCourses.length === this.allCheckedCourses.length;
```
Similarly for the rest below.

Reply:
I couldnt do this as `this.allPresentCourses.length` is not a boolean, which gives me compilation error

Reply:
Why would it give you compile-time error? JS/TS supports boolean-coercion.

Reply:
I got something like this:

![image](https://user-images.githubusercontent.com/28526038/52424934-f5496600-2b35-11e9-87b3-e91f0e3f25e8.png)

Reply:
Will a little help of `!!` work?
```ts
return !!this.allPresentCourses.length && this.allPresentCourses.length === this.allCheckedCourses.length;
```